### PR TITLE
feat: setup kubernetes job for populating db policies and sample data

### LINKED ISSUE #78: feat: setup kubernetes job for populating db policies and sample data
The task is to set up a K8s job to set up db policies and populate sample data. This will prevent port-forwarding/exposing the database container which is potentially risky and should not be done for database services.

### LINKED ISSUE #83: build: configure github action to PYPI [TEST] deployment for support_sphere_py
This will publish a package on test PYPI for support_sphere_py

Usage: `import support_sphere`

### PR DESCRIPTION
**Pull request by anujsinha3**
Fixes #78 
Fixes #83

### CODE REVIEW

File: src/support_sphere_py/tests/resources/scripts/update_db_sample_data.py [hunk 1]
```diff
@@ -72,11 +72,12 @@ def populate_cluster_and_household_details():
 
 def get_supabase_client() -> Client:
 
+    logger.info("Establishing connection via supabase")
     # Setting up the supabase client for python
     file_path = Path("./deployment/values.dev.yaml")
     with file_path.open(mode='r') as file:
         config = yaml.safe_load(file)
-        url = config['studio']['environment']['SUPABASE_PUBLIC_URL']
+        url = 'http://supabase-supabase-kong:8000/'
```
Comment (lsetiawan) [inline src/support_sphere_py/tests/resources/scripts/update_db_sample_data.py:80]: This should still take the regular supabase public url.
```suggestion
        url = os.environ.get("SUPABASE_PUBLIC_URL")
```

**Review by lsetiawan — CHANGES_REQUESTED** [review]
We can go over this today to get it to the correct setup.

File: deployment/docker/populate_db.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+# Define ARGs for environment variables that will be passed at build time
+ARG DB_HOST
+ARG DB_PORT
+ARG DB_USERNAME
+ARG DB_PASSWORD
+ARG DB_NAME
+ARG SUPABASE_KONG_HOST
+ARG SUPABASE_KONG_PORT
+ARG JWT_ANON_KEY
```
Comment (lsetiawan) [inline deployment/docker/populate_db.Dockerfile:17]:
```suggestion
```
For an app level environment, you won't need ARGS

File: deployment/docker/populate_db.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+# Below ENV values are overridden when running the via K8s Job else the above arg values are used.
+ENV DB_HOST=${DB_HOST}
+ENV DB_PORT=${DB_PORT}
+ENV DB_USERNAME=${DB_USERNAME}
+ENV DB_PASSWORD=${DB_PASSWORD}
+ENV DB_NAME=${DB_NAME}
+ENV SUPABASE_KONG_HOST=${SUPABASE_KONG_HOST}
+ENV SUPABASE_KONG_PORT=${SUPABASE_KONG_PORT}
+ENV JWT_ANON_KEY=${JWT_ANON_KEY}
```
Comment (lsetiawan) [inline deployment/docker/populate_db.Dockerfile:27]:
```suggestion
ENV DB_HOST=localhost
ENV DB_PORT=5432
ENV DB_USERNAME=postgres
ENV DB_PASSWORD=example123456
ENV DB_NAME=postgres
ENV SUPABASE_PUBLIC_URL=http://localhost/
ENV JWT_ANON_KEY=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyAgCiAgICAicm9sZSI6ICJhbm9uIiwKICAgICJpc3MiOiAic3VwYWJhc2UtZGVtbyIsCiAgICAiaWF0IjogMTY0MTc2OTIwMCwKICAgICJleHAiOiAxNzk5NTM1NjAwCn0.dc_X5iR_VP_qT0zsiyj_I_OZ2T9FtRU2BBNWN8Bu4GE
```

Comment (lsetiawan) [reply deployment/docker/populate_db.Dockerfile:27]: Use similar defaults as the dev yaml

File: deployment/kubernetes/populate_db_job.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+          - name: SUPABASE_KONG_PORT
+            valueFrom:
+              configMapKeyRef:
+                name: config-dev
+                key: SUPABASE_KONG_PORT
+          - name: JWT_ANON_KEY
+            valueFrom:
+              secretKeyRef:
+                name: secret-dev
+                key: JWT_ANON_KEY
```
Comment (lsetiawan) [inline deployment/kubernetes/populate_db_job.yaml:53]:
```suggestion
                name: supabase-jwt
                key: anonKey
```

This already exists

File: deployment/kubernetes/populate_db_job.yaml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+          - name: DB_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                name: secret-dev
+                key: DB_PASSWORD
+          - name: DB_NAME
+            valueFrom:
+              configMapKeyRef:
+                name: config-dev
+                key: DB_NAME
```
Comment (lsetiawan) [inline deployment/kubernetes/populate_db_job.yaml:38]:
```suggestion
              secretKeyRef:
                name: supabase-db
                key: database
```

File: deployment/kubernetes/populate_db_job.yaml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+          - name: DB_USERNAME
+            valueFrom:
+              configMapKeyRef:
+                name: config-dev
+                key: DB_USERNAME
+          - name: DB_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                name: secret-dev
+                key: DB_PASSWORD
```
Comment (lsetiawan) [inline deployment/kubernetes/populate_db_job.yaml:33]:
```suggestion
              secretKeyRef:
                name: supabase-db
                key: password
```

File: deployment/kubernetes/populate_db_job.yaml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+          - name: DB_PORT
+            valueFrom:
+              configMapKeyRef:
+                name: config-dev
+                key: DB_PORT
+          - name: DB_USERNAME
+            valueFrom:
+              configMapKeyRef:
+                name: config-dev
+                key: DB_USERNAME
```
Comment (lsetiawan) [inline deployment/kubernetes/populate_db_job.yaml:28]:
```suggestion
              secretKeyRef:
                name: supabase-db
                key: username
```

File: deployment/kubernetes/populate_db_job.yaml [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+      containers:
+      - name: populate-db-job
+        image: pdc/populate_db:1.0
+        imagePullPolicy: Never
+        env:
+          - name: DB_HOST
+            valueFrom:
+              configMapKeyRef:
+                name: config-dev
+                key: DB_HOST
```
Comment (lsetiawan) [inline deployment/kubernetes/populate_db_job.yaml:18]:
```suggestion
          - name: DB_HOST
            value: supabase-supabase-db
```

File: deployment/kubernetes/populate_db_job.yaml [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+          - name: DB_PORT
+            valueFrom:
+              configMapKeyRef:
+                name: config-dev
+                key: DB_PORT
```
Comment (lsetiawan) [inline deployment/kubernetes/populate_db_job.yaml:23]:
```suggestion
          - name: DB_PORT
            value: "5432"
```

File: deployment/kubernetes/populate_db_job.yaml [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+          - name: DB_NAME
+            valueFrom:
+              configMapKeyRef:
+                name: config-dev
+                key: DB_NAME
+          - name: SUPABASE_KONG_HOST
+            valueFrom:
+              configMapKeyRef:
+                name: config-dev
+                key: SUPABASE_KONG_HOST
```
Comment (lsetiawan) [inline deployment/kubernetes/populate_db_job.yaml:43]:
```suggestion
          - name: SUPABASE_KONG_HOST
            value: supabase-supabase-kong
```

File: deployment/kubernetes/populate_db_job.yaml [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+          - name: SUPABASE_KONG_PORT
+            valueFrom:
+              configMapKeyRef:
+                name: config-dev
+                key: SUPABASE_KONG_PORT
```
Comment (lsetiawan) [inline deployment/kubernetes/populate_db_job.yaml:48]:
```suggestion
          - name: SUPABASE_KONG_PORT
            value: "8000"
```

**Review by lsetiawan — CHANGES_REQUESTED** [review]
I think you're really close here. A lot of the values already exists within the kubernetes cluster. You'll just need to use them, especially secrets.

File: deployment/kubernetes/populate_db_job.yaml (hunk 11 above)
Comment (anujsinha3) [reply deployment/kubernetes/populate_db_job.yaml:48]: @lsetiawan 
Just to confirm, for these cases, where the values are not pre-existing in the cluster configmap, do you want me to hard-code those in the `job.yaml`?


Yesterday, I believe we were more inclined towards passing these values through the `kubectl` command line like below for flexibility, right?

```
kubectl create configmap config-dev \
  --from-literal=DB_HOST="supabase-supabase-db" \
  --from-literal=DB_PORT="5432" \
  --from-literal=SUPABASE_KONG_HOST="supabase-supabase-kong" \
  --from-literal=SUPABASE_KONG_PORT="8000"
```

File: src/support_sphere_py/src/support_sphere/repositories/__init__.py
Comment (lsetiawan) [inline src/support_sphere_py/src/support_sphere/repositories/__init__.py:1]: Great place for this!

**Review by lsetiawan — APPROVED** [review]
This looks good. There are definitely many moving parts here, hopefully that the cd works this time, at which we'll probable need to update the dockerfile. Or include a docker image within the CD... but that's a separate issue. Thanks.

**Pull request merged**